pkg/util/machineset/util_test.go

```diff
@@ -0,0 +1,200 @@
+package util
+
+import (
+	"maps"
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestSettingAnnotations(t *testing.T) {
+	tests := []struct {
+		name                string
+		value               string
+		fn                  func(map[string]string, string) map[string]string
+		expectedAnnotations map[string]string
+		suppliedAnnotations map[string]string
+	}{
+		{
```

Comment:
let's make sure we have a test case that sets an empty string so that we know it works

Reply:
will do!